### PR TITLE
*: support sig push down in mocktikv.

### PR DESCRIPTION
**Pull request by hanfei1991**

### CODE REVIEW

File: expression/expr_to_pb.go [hunk 1]
```diff
@@ -491,5 +491,11 @@ func (pc pbConverter) convertToPBExpr(expr *ScalarFunction, tp tipb.ExprType) *t
 		}
 		children = append(children, pbArg)
 	}
+	if pc.client.IsRequestTypeSupported(kv.ReqTypeDAG, kv.ReqSubTypeSignature) {
+		code := expr.Function.PbCode()
+		if code > 0 {
+			return &tipb.Expr{Tp: tipb.ExprType_ScalarFunc, Sig: &code, Children: children, FieldType: toPBFieldType(expr.RetType)}
```
Comment (zz-jason) [inline expression/expr_to_pb.go]: `code` is a `int32` value, it's not suitable to escape this local variable to be allocated on heap and keep a reference of it. we can change the pb or let `Sig` point to a global variable.

Comment (hanfei1991) [reply expression/expr_to_pb.go]: you are right. I will change this to a value in another pr

File: expression/builtin_arithmetic.go [hunk 2]
```diff
@@ -497,13 +497,11 @@ func (c *arithmeticDivideFunctionClass) getFunction(args []Expression, ctx conte
 		bf := newBaseBuiltinFuncWithTp(args, ctx, tpReal, tpReal, tpReal)
 		c.setType4DivReal(bf.tp)
 		sig := &builtinArithmeticDivideRealSig{baseRealBuiltinFunc{bf}}
-		sig.setPbCode(tipb.ScalarFuncSig_MultiplyReal)
```
Comment (XuHuaiyu) [inline expression/builtin_arithmetic.go]: forget to setPbCode(ipb.ScalarFuncSig_DivReal) ?

File: expression/builtin_arithmetic.go [hunk 3]
```diff
@@ -497,13 +497,11 @@ func (c *arithmeticDivideFunctionClass) getFunction(args []Expression, ctx conte
...
 		return sig.setSelf(sig), nil
 	}
 	bf := newBaseBuiltinFuncWithTp(args, ctx, tpDecimal, tpDecimal, tpDecimal)
 	c.setType4DivDecimal(bf.tp, tpA, tpB)
 	sig := &builtinArithmeticDivideDecimalSig{baseDecimalBuiltinFunc{bf}}
-	sig.setPbCode(tipb.ScalarFuncSig_MultiplyDecimal)
```
Comment (XuHuaiyu) [inline expression/builtin_arithmetic.go]: ditto

File: expression/builtin_arithmetic.go (hunk 2 above)
Comment (hanfei1991) [reply expression/builtin_arithmetic.go]: There is not this sig in pb

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**